src/utils/format.utils.ts

```diff
@@ -47,3 +47,165 @@ export const formatMoney = (number: number, options?: FormatMoneyOptions) => {
         return number.toString();
     }
 };
+
+/**
+ * Converts and formats a given date input into a specified string format without usage of external libraries such as moment.js or date-fns.
+ *
+ * @param {Date | number | string} dateInput - The date to be formatted.
+ *   - Can be a Date object, Unix timestamp, or date string.
+ * @param {Intl.DateTimeFormatOptions} [options] - Optional configuration for date formatting.
+ *   - Defaults to { day: "2-digit", month: "2-digit", year: "numeric" }.
+ * @param {string} [format="YYYY-MM-DD"] - The desired output format.
+ *   - Supported formats: 'YYYY-MM-DD', 'DD MMM YYYY', 'MMM DD YYYY'.
+ * @param {boolean} [unix=false] - If true, treats the numeric input as a Unix timestamp.
+ *
+ * @returns {string} A formatted date string according to the specified format.
+ *
+ * @example
+ * // Returns date in 'YYYY-MM-DD' format
+ * getFormattedDateString(new Date('2023-05-15'));
+ * // => "2023-05-15"
+ *
+ * @example
+ * // Returns date in 'DD MMM YYYY' format
+ * getFormattedDateString('2023-05-15', {}, 'DD MMM YYYY');
+ * // => "15 May 2023"
+ */
+export const getFormattedDateString = (
+    dateInput: Date | number | string,
+    options: Intl.DateTimeFormatOptions = { day: "2-digit", month: "2-digit", year: "numeric" },
```

Comment:
[SUGGESTION] Can we encapsulate `options`, `format`, `unix` into a single object, similar to how we handle `formatMoney` util? This approach would simplify the process for users by allowing them to pass only the necessary properties. 🤔